src/classes/RD2_OpportunityService.cls

```diff
@@ -256,11 +258,8 @@ public with sharing class RD2_OpportunityService {
             RD2_ScheduleService.Installment earliestUnmatchedInstallment;
 
             for (RD2_OpportunityMatcher.Record record : matcher.getRecords()) {
-                isOpenOppMatched = false;
 
-                if (record.hasInstallment() 
-                    && (record.isNew() || !record.getInstallment().isSkipped)
-                ) {
+                if (record.hasInstallment() && record.isNew()) {
```

Comment:
I noticed that in this method, the previous logic to check for a skipped installment has been removed. I can't find where that logic still executes elsewhere though. Is the premise here that you no longer check for a paused installment in the opp service - which doesn't seem right here.

Reply:
That's a valid point and I planned to indicate more information about code changes, did not have time.
The `isSkipped` is not required to be checked since we get active installments only from `getVisualizedInstallments()`. The `isSkipped` installment checks bring complexity in the conditional statements and it is applicable only if we get all installment instead ie `getAllVisualizedInstallments()`. So instead of checking which conditional statement is causing the error, it was cleaner to remove all `isSkipped` checks in the method. Let me know if you suggest different approach.